From 0a6ebc6a9b4b580ffe32ec58bdea20485b6b6cdf Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <skardach@marvell.com>
Date: Thu, 18 Apr 2019 11:28:10 +0200
Subject: [PATCH 158/386] octeontx2-rm: fix strncpy ranges

Fix potential buffer overflow on copying domain name passed by the user.

Change-Id: Ic57ae2e87fcbf1919f91a8b1c0d58b19a670dc39
Signed-off-by: Stanislaw Kardach <skardach@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7810
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-rm/domain_sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-rm/domain_sysfs.c b/drivers/soc/marvell/octeontx2-rm/domain_sysfs.c
index 55144354c406..7e2193f3222d 100644
--- a/drivers/soc/marvell/octeontx2-rm/domain_sysfs.c
+++ b/drivers/soc/marvell/octeontx2-rm/domain_sysfs.c
@@ -212,7 +212,7 @@ do_create_domain(struct domain_sysfs *lsfs, struct domain_params *dparams)
 		res = -ENODEV;
 		goto err_dom;
 	}
-	strncpy(domain->name, dparams->name, DOMAIN_NAME_LEN);
+	strncpy(domain->name, dparams->name, DOMAIN_NAME_LEN - 1);
 	domain->in_use = true;
 	/* Verify ports are valid and supported. */
 	if (dparams->port_cnt == 0)
@@ -460,7 +460,7 @@ destroy_domain_store(struct kobject *kobj, struct kobj_attribute *attr,
 	char name[DOMAIN_NAME_LEN], *name_ptr;
 	int i, res;
 
-	strncpy(name, buf, DOMAIN_NAME_LEN);
+	strncpy(name, buf, DOMAIN_NAME_LEN - 1);
 	name_ptr = strim(name);
 	if (strlen(name_ptr) == 0) {
 		dev_err(dev, "Empty domain name.\n");
-- 
2.17.1

